### ISSUE TITLE
Support `clamp()` with tensor min and max

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#2793]
It might be useful to extend the current `clamp()` function, where the min and max can be tensor. 

e.g. To clip the value for a [5, 3]-tensor, we could define min and max for each element, or each row/column. 

Finally, it might be good to rename it to clip to be consistent with `numpy.clip`
 

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @mruberry @rgommers @heitorschueroff

**Comment by nishnik** [comment]
Hey @zuoxingdong, how to approach this.
I would like to contribute

**Comment by GuillaumeLeclerc** [comment]
I think it would ideed be a good thing. I did not find a better way than: `torch.stack` and then a clamp

**Comment by yaceben** [comment]
any update on this ?

**Comment by adam-dziedzic** [comment]
@yaceben thanks for the one-liner

**Comment by dniku** [comment]
One could also use `torch.where`, similar to the way suggested in [this thread](https://discuss.pytorch.org/t/how-to-threshold-each-channel-in-a-tensor-selectively/43612).

**Comment by devpouya** [comment]
Any progress here?

**Comment by arpanmukherjee** [comment]
I can take up this issue if anyone hasn't.

**Comment by mruberry** [comment]
I've updated the labels on this function. We would still accept a PR implementing this behavior. Note that this functionality is consistent with NumPy, where np.clip can accept array-likes.

**Comment by mruberry** [comment]
Updating to high priority based on user activity.

**Comment by ssgosh** [comment]
This feature will be very useful for clipping multi-channel images with different ranges for each channel.
